src/pykeen/models/nbase.py

```diff
@@ -498,19 +519,24 @@ def _repeat_scores_if_necessary(
 
         return scores.repeat(*repeats)
 
+    def _entity_representation_from_mode(self, mode: Mode = None):
+        return self.entity_representations
+
     def _get_representations(
         self,
         h_indices: Optional[torch.LongTensor],
         r_indices: Optional[torch.LongTensor],
         t_indices: Optional[torch.LongTensor],
+        mode: Mode = None,
     ) -> Tuple[HeadRepresentation, RelationRepresentation, TailRepresentation]:
         """Get representations for head, relation and tails, in canonical shape."""
+        entity_representations = self._entity_representation_from_mode(mode=mode)
         h, r, t = [
             [representation.get_in_more_canonical_shape(dim=dim, indices=indices) for representation in representations]
             for dim, indices, representations in (
-                ("h", h_indices, self.entity_representations),
+                ("h", h_indices, entity_representations),
                 ("r", r_indices, self.relation_representations),
```

Comment:
should `relation_representations` ever be different based on mode?

Reply:
No, the same motivation as in the previous comment